New Model for Employee, Update old project model, etc.

TarikVelic opened the pull request:
--------------------------------------------------------------------
 MAIN
 * Added new Model for Employee
 * Updated the old project model so that it looks for an employee instead of a user
 * Deleted some things from the user model, service and etc that we dont need when employee model is there --------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

TarikVelic committed:
New Model for Employee, Update old project model, etc.

--------------------------------------------------------------------
 MAIN
 * Added new Model for Employee
 * Updated the old project model so that it looks for an employee instead of a user
 * Deleted some things from the user model, service and etc that we dont need when employee model is there
--------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

TarikVelic committed:
Added get for pagination, search query, page, limit etc.

becarselman commented on controllers/employee.js:
```diff
@@ -0,0 +1,61 @@
+exports.getEmployeeById = async (req, res) => {
+  try {
+    const employee = await employeeService.getEmployeeById(req.params.id);
+    if (!employee) {
```
You will do this in service. You can check this and throw error for this case

becarselman commented on db/models/employee.js:
```diff
@@ -0,0 +1,32 @@
+  },
+  department: {
+    type: String,
+    enum: ['Management', 'Administration', 'Design', 'Development'],
```
can we specify this enum variables in some utils file for example utils/enums where will be all enums

TarikVelic committed:
Problems from review fixed + added route to get all employees without using pagination route

becarselman commented on controllers/employee.js:
```diff
@@ -0,0 +1,71 @@
+exports.getAllEmployees = async (req, res) => {
+  try {
+    const employees = await employeeService.getAllEmployees();
+    res.json(employees);
```
You are missing status

becarselman commented on controllers/employee.js:
```diff
@@ -0,0 +1,71 @@
+exports.getEmployeeById = async (req, res) => {
+  try {
+    const employee = await employeeService.getEmployeeById(req.params.id);
+    res.json(employee);
```
you are missing status code

becarselman commented on controllers/employee.js:
```diff
@@ -0,0 +1,71 @@
+exports.createEmployee = async (req, res) => {
+  try {
+    const employee = await employeeService.createEmployee(req.body);
+    res.status(201).json(employee);
```
Also you are missing return here. You need to return this and on all places similar to this

becarselman commented on controllers/employee.js:
```diff
@@ -0,0 +1,71 @@
+exports.updateEmployee = async (req, res) => {
+  try {
+    const employee = await employeeService.updateEmployee(req.params.id, req.body);
+    if (!employee) {
```
If I leave comment on one place please correct on all places where you did same mistake

becarselman commented on repositories/employee.js:
Where is pagination here.

TarikVelic committed:
Fixed repositories and service

TarikVelic committed:
FEATURE/TariikMVP/Update

--------------------------------------------------------------------
 MAIN
 * Small fixes
--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

becarselman commented on controllers/projects.js:
```diff
@@ -1,9 +1,25 @@
+      for (const developer of developers) {
+        const employee = await employeeService.getEmployeeById(developer.employee);
+        if (!employee) {
+          return res.status(400).json({ error: 'Employee not found.' });
```
you can reuse error message not found

becarselman commented on controllers/projects.js:
```diff
@@ -1,9 +1,25 @@
+      const { developers, ...projectData } = req.body;
+      const projectDevelopers = [];
+
+      for (const developer of developers) {
```
this need to be done in service

becarselman commented on repositories/employee.js:
```diff
@@ -0,0 +1,63 @@
+class EmployeeRepository {
+  async createEmployee(employeeData) {
+    try {
+      const employee = new Employee(employeeData);
```
use orm

becarselman commented on services/employee.js:
```diff
@@ -0,0 +1,53 @@
+const employeeRepository = require('../repositories/employee');
+
+class EmployeeService {
```
why you start using class if we use functions before. Please stay consistent

TarikVelic committed:
Bug fixes

TarikVelic committed:
Deleted !employee from controller

becarselman reviewed (approved):
I will approve this, But I have additional questions please ping me over discord

TarikVelic committed:
Service to get only firstName, lastName and id

becarselman commented on services/employee.js:
```diff
@@ -0,0 +1,81 @@
+  async getAllEmployees() {
+    try {
+      const employees = await employeeRepository.getAllEmployees();
+      const simplifiedEmployees = employees.map((employee) => {
```
this is not how we do this. Please use orm and do it in repository. This approach is same as before. We need to do this on db call and not here. https://stackoverflow.com/questions/67809187/mongoose-how-to-return-only-specific-field

TarikVelic replied:
fixed

TarikVelic committed:
Update for repositories so getAllEmployees can get only id, firstName and lastName
